config/routes.rb

```diff
@@ -11,7 +11,10 @@
 
   # Defining the application routes for the user posts in the application
   resources :users, only: [:index,:show] do 
-  resources :posts, only: [:index,:show]
+  resources :posts, only: [:index,:show,:new,:create] do
+  resources :comments,only: [:new, :create]
+  resources :likes,only: [:new, :create]
```

Comment:
- **[OPTIONAL]** Kindly remove the `:new` route for likes here, based on the other feedbacks.